lighthouse-core/lib/asset-saver.js

```diff
@@ -198,9 +198,30 @@ function saveAssets(artifacts, audits, pathWithBasename) {
   });
 }
 
+/**
+ * Log trace(s) and associated screenshot(s) to console.
+ * @param {!Artifacts} artifacts
+ * @param {!Audits} audits
+ * @return {!Promise}
+ */
+function logAssets(artifacts, audits) {
+  return prepareAssets(artifacts, audits).then(assets => {
+    const data = assets[0];
+    log.log('devtoolslog.json', JSON.stringify(data.devtoolsLog, null, 2));
+    log.log('screenshots.html', data.screenshotsHTML);
+    const traceIter = traceJsonGenerator(data.traceData);
+    let traceJson = '';
+    for (const trace of traceIter) {
+      traceJson += trace;
```

Comment:
This is a tricky one. We use the traceJsonGenerator to avoid constructing strings that are 256MB. 

However [that 256mb limit was bumped to 1gb in Chrome 62](https://bugs.chromium.org/p/v8/issues/detail?id=6148) so we can remove it sometime soon. Given that, it seems fine for us to just concat the string like this. 👍